src/components/DatePickerSingle.react.js

```diff
@@ -53,24 +56,19 @@ export default class DatePickerSingle extends Component {
 
     isOutsideRange(date) {
         const {min_date_allowed, max_date_allowed} = this.state;
-        const notUndefined = R.complement(
-            R.pipe(
-                R.type,
-                R.equals('Undefined')
-            )
-        );
+
         return (
-            (notUndefined(min_date_allowed) && date < min_date_allowed) ||
-            (notUndefined(max_date_allowed) && date >= max_date_allowed)
+            (min_date_allowed && date.isBefore(min_date_allowed)) ||
+            (max_date_allowed && date.isAfter(max_date_allowed))
         );
     }
 
     onDateChange(date) {
         const {setProps} = this.props;
-        if (setProps && date !== null) {
+
+        this.setState({date});
+        if (date) {
```

Comment:
`setProps` check is now usless

Reply:
This whole `if` may be useless, if we accept #511

Reply:
Yes! I saw the PR :)